packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -399,6 +416,20 @@ export class ProfilesCache {
         return allTypes;
     }
 
+    private getParentProfileForToken(profileName: string, config: imperative.Config): string {
+        const secureProps = config.api.secure.secureFields();
+        let parentProfile = profileName.slice(0, profileName.lastIndexOf("."));
+        let tempProfile = profileName;
+        while (tempProfile.includes(".")) {
+            tempProfile = tempProfile.slice(0, tempProfile.lastIndexOf("."));
+            if (secureProps.includes(`${config.api.profiles.getProfilePathFromName(tempProfile)}.properties.tokenValue`)) {
```

Comment:
Nice way to handle complex nesting scenarios.

Seems like this works even for a token value stored in the QA level of `QA.Plex1.Lpar1.zosmf` 😋